crates/wasm-smith/src/core.rs

```diff
@@ -1424,6 +1424,7 @@ impl Module {
         Ok(GlobalType {
             val_type: self.arbitrary_valtype(u)?,
             mutable: u.arbitrary()?,
+            shared: self.config.shared_everything_threads_enabled && u.arbitrary()?,
```

Comment:
FWIW I'd recommend leaving wasm-smith out of the equation entirely rather than piecemeal updating it. I find it's best to do wasm-smith as a dedicated project once validation is done to fuzz the two in tandem. 

For example wasm-smith is going to need a lot of updates to avoid accessing non-shared items from a shared context, so while this precise change may remain there's going to need to be a fair bit of infrastructure elsewhere too. 

In the meantime it's ok to ignore wasm-smith entirely while you're first implementing the shared-everything proposal.